src/main/java/formflow/library/pdf/PdfFile.java

```diff
@@ -14,37 +15,44 @@
 @Slf4j
 public record PdfFile(String path, String name) {
 
-    public static PdfFile copyToTempFile(String path) {
-        InputStream unfilledPdf = PdfFile.class.getResourceAsStream(path);
-        int fileExtension = path.length() - 4;
-        File tempFile;
-        Path pathToTempFile;
-        try {
-            tempFile = File.createTempFile(path, path.substring(fileExtension), new ApplicationTemp().getDir());
-            pathToTempFile = Files.write(tempFile.toPath(), unfilledPdf.readAllBytes());
-        } catch (NullPointerException | IOException e) {
-            throw new RuntimeException("Could not copy pdf resource to temp file: " + e);
-        }
-        return new PdfFile(pathToTempFile.toAbsolutePath().toString(), getNameFromPath(path));
-    }
-
-    private static String getNameFromPath(String path) {
-        return path.substring(path.lastIndexOf('/') + 1, path.length() - 4);
-    }
-
-    @Override
-    public String toString() {
-        return path;
-    }
-
-    public byte[] fileBytes() throws IOException {
-        return Files.readAllBytes(Paths.get(this.path));
-    }
-
-    public void finalizeForSending() throws IOException {
-        PDDocument pdDocument = PDDocument.load(fileBytes());
-        pdDocument.getDocumentCatalog().getAcroForm().flatten();
-        pdDocument.save(path);
-        pdDocument.close();
+  // put all the file in the same tmp java location
+  private static final File tmpFileDir = new ApplicationTemp().getDir();
+
+  public static PdfFile copyToTempFile(String pathToResource) {
+    InputStream unfilledPdf = PdfFile.class.getResourceAsStream(pathToResource);
+    Path resourcePath = Path.of(pathToResource);
+    String fileNameWithExtension = resourcePath.getFileName().toString();
+    String fileExtension = fileNameWithExtension.substring(fileNameWithExtension.lastIndexOf('.'));
```

Comment:
I didn't want to make assumptions about file extension, so I moved a few things around here.  Also just made it take the fileName and fileExtension as parameters to the createFileTemp. It makes for a cleaner temp file: 

`Multipage-UBI-Form3307367797630564204.pdf`

Versus: 

`testPdf.pdf17812849852771651595.pdf`